File: src/main/java/com/IconPippi/simtriggers/data/request/DataRequestDecoder.java
package com.IconPippi.simtriggers.data.request;

/**
 * This class (similar to EventDecoder) decodes data definitions IDs and data request IDs into strings
 * @author IconPippi
 * 
 */
public class DataRequestDecoder {
	
	/**
	 * Decode a data definition's ID
	 * @param dataDefinitionID input ID
	 * @return decoded string
	 */
	public String decodeDataDefinitionID(int dataDefinitionID) {
		String toReturn = null;
		int count = 0;
		
		for (int i : DataRequest.dataDefinitionIDs) {
			count++;
			if (i == dataDefinitionID) {
				toReturn = DataRequest.dataDefinitions.get(count-1);
				break;
			} else {
				continue;
			}
		}
		return toReturn;
	}
	
	/**
	 * Decode a data request's ID
	 * @param requestID input requestID
	 * @return decoded string
	 */
	public String decodeRequestID(int requestID) {
		String toReturn = null;
		int count = 0;
		
		for (int i : DataRequest.requestIDs) {
			count++;
			if (i == requestID) {
				toReturn = DataRequest.requests.get(count-1);
				break;
			} else {
				continue;
			}
		}
		return toReturn;
	}
}

File: src/main/kotlin/com/simtriggers/fsx/data/request/DataRequestDecoder.kt
package com.simtriggers.fsx.data.request

/**
 * 06/08/2019
 * Decode the received data
 *
 * @author IconPippi
 */
class DataRequestDecoder {

    /**
     * Decode a data definition's ID
     * @param dataDefinitionID Input ID
     * @return Decoded string
     */
    fun decodeDataDefinitionID(dataDefinitionID: Int): String? {
        var toReturn: String? = null
        var count = 0

        for (i in RequestData.dataDefinitionIDs) {
            count++
            if (i == dataDefinitionID) {
                toReturn = RequestData.dataDefinitions[count - 1]
                break
            } else {
                continue
            }
        }
        return toReturn
    }

    /**
     * Decode a data request's ID
     * @param requestID Input requestID
     * @return Decoded string
     */
    fun decodeRequestID(requestID: Int): String? {
        var toReturn: String? = null
        var count = 0

        for (i in RequestData.requestIDs) {
            count++
            if (i == requestID) {
                toReturn = RequestData.requests[count - 1]
                break
            } else {
                continue
            }
        }
        return toReturn
    }

}